newrelic/resource_newrelic_synthetics_monitor.go

```diff
@@ -95,158 +94,411 @@ func resourceNewRelicSyntheticsMonitor() *schema.Resource {
 				Optional:    true,
 				Description: "Fail the monitor check if redirected.",
 			},
+			"runtime_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "",
+			},
+			"runtime_type_version": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "",
+			},
+			"script_language": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "",
+			},
+			"tags": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "",
+						},
+						"values": {
+							Type:        schema.TypeList,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Required:    true,
+							Description: "",
+						},
+					},
+				},
+			},
+			"enable_screenshot_on_failure_and_script": {
+				Type:        schema.TypeBool,
+				Description: "",
+				Optional:    true,
+			},
+			"custom_headers": {
+				Type:        schema.TypeSet,
+				Description: "",
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Description: "",
+							Optional:    true,
+						},
+						"value": {
+							Type:        schema.TypeString,
+							Description: "",
+							Optional:    true,
+						},
+					},
+				},
+			},
 		},
 	}
 }
 
-func buildSyntheticsMonitorStruct(d *schema.ResourceData) synthetics.Monitor {
-	monitor := synthetics.Monitor{
-		Name:         d.Get("name").(string),
-		Type:         synthetics.MonitorType(d.Get("type").(string)),
-		Frequency:    uint(d.Get("frequency").(int)),
-		Status:       synthetics.MonitorStatusType(d.Get("status").(string)),
-		SLAThreshold: d.Get("sla_threshold").(float64),
+//func to build the input to create simple browser monitor
+func buildSyntheticsSimpleBrowserMonitor(d *schema.ResourceData) synthetics.SyntheticsCreateSimpleBrowserMonitorInput {
+	inputBase := expandSyntheticsMonitorBase(d)
+
+	simpleBrowserMonitorInput := synthetics.SyntheticsCreateSimpleBrowserMonitorInput{}
```

Comment:
This is a small nitpick, but why didn't we keep the initialisation as it was before?

```
monitor := synthetics.Monitor{
		Name:         d.Get("name").(string),
		Type:         synthetics.MonitorType(d.Get("type").(string)),
		Frequency:    uint(d.Get("frequency").(int)),
```
vs this